### PR TITLE
fix: add CORS

### PR DESCRIPTION
**Pull request by fubuloubu**

### CODE REVIEW

**Comment by fubuloubu** [comment]
@Ninjagod1251 please evaluate locally

File: main.py [hunk 1]
```diff
@@ -16,6 +17,20 @@
 
 PackageManifest.update_forward_refs()
 app = FastAPI()
+app.add_middleware(
```
Comment (fubuloubu) [inline main.py:20]: not sure if we want to shut off anything in prod, verify with remix

File: main.py [hunk 2]
```diff
@@ -16,6 +17,20 @@
...
+    CORSMiddleware,
+    allow_origins=[
+        # NOTE: When running `npm run dev`, this is the URL
+        "http://localhost:5173",
+        # NOTE: When running `npm run build && npm run preview`, this is the URL
+        "http://localhost:4173",
+        # NOTE: This is where the UI gets hosted
+        "https://remix.ethereum.org",
```
Comment (yann300) [inline main.py:28]: could we also add
https://remix-alpha.ethereum.org
https://remix-beta.ethereum.org
package://6fd22d6fe5549ad4c4d8fd3ca0b7816b.mod (this is for the desktop version)
thanks!

Comment (fubuloubu) [reply main.py:28]:
```suggestion
        "https://remix.ethereum.org",
        "https://remix-alpha.ethereum.org",
        "https://remix-beta.ethereum.org",
```

File: main.py [hunk 3]
```diff
@@ -16,6 +17,22 @@
 
 PackageManifest.update_forward_refs()
 app = FastAPI()
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=[
+        # NOTE: When running `npm run dev`, this is the URL
+        "http://localhost:5173",
```
Comment (fubuloubu) [inline main.py:24]:
```suggestion
        # NOTE: When running remix in local dev, this is the URL
        "http://localhost:8080",
```

**Pull request merged**